ec2-cluster/modules/node/main.tf

```diff
@@ -0,0 +1,198 @@
+resource "aws_instance" "node" {
+  count = var.instance_count
+
+  ami                         = data.aws_ami.node.id
+  instance_type               = var.instance_type
+  associate_public_ip_address = true
+
+  root_block_device {
+    volume_type = "gp2"
+    volume_size = var.disk_size
+  }
+
+  user_data = file("ssh_keys.sh")
+
+  tags = {
+    Index     = count.index
+    Name      = "node-${var.cluster_id}-${count.index}"
+    Env       = var.env
+    Cluster   = var.cluster_id
+    Workspace = var.workspace
+  }
+
+  connection {
+    host    = self.public_ip
+    type    = "ssh"
+    user    = local.ssh_user
+    timeout = "240s"
+  }
+
+  provisioner "file" {
+    content = templatefile("${path.module}/templates/setup", {
+      tessellation_version = var.tessellation_version,
+      key                  = var.instance_keys[count.index].key
+      app_env              = var.env
+    })
+    destination = "/tmp/setup"
+  }
+
+  provisioner "file" {
+    source      = "${path.module}/templates/setup-services"
+    destination = "/tmp/setup-services"
+  }
+
+  provisioner "file" {
+    source      = "${path.module}/data/key-${count.index}.p12"
```

Comment:
Where these files come from? Maybe we shouldn't copy the `.p12` keys via the script at all.